modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
+
+    returns:  The longest common prefix path of all the input paths
+    */
+    proc commonPrefix(paths: []): string {
+
+    var prefix: string = "";    // result string
+    var inputLength = paths.size;   // size of input array
+    var firstPath = paths[1];
```

Comment:
This assumes that the array provided is 1's based.  However, that may not be the case, as Chapel arrays can start at any index.  You could also be provided a slice of an array.
The array's domain has a method `first` that you can use like this:
```chapel
var firstPath = paths[paths.domain.first];
```

Since this will impact other parts of your algorithm, I recommend looking at the overall correctness in other places first, then coming back and extending the function in this way

Reply:
(Note also that the varargs version will always have the same index set, I believe)

Reply:
Thank you for mentioning this feature. We have changed the code accordingly for arrays